### PR TITLE
Test for cardboard list callout

### PR DESCRIPTION
**Pull request by iremgokce**
### Summary of changes 🔍 
Added test file for `CardboardListCallout` component for exposing the search term on its primary action click
image

### Checklist ✔️
- [ ] Linked associated issue (if present)
- [x] Added relevant labels
- [x] Requested developer reviews
- [ ] Request UI review from design / PM
- [x] Verify all code checks are passing

### CODE REVIEW

File: jest.setup.ts [hunk 1]
```diff
@@ -1,5 +1,15 @@
 import '@testing-library/jest-dom/extend-expect';
 import 'jest-canvas-mock';
+import i18n from './src/i18n';
 
 jest.mock('./src/i18n.ts', () => ({ t: (key: string) => key }));
+jest.mock('react-i18next', () => ({
```
Comment (msnyder-msft) [inline jest.setup.ts:6]: yessss nice find this is perfect.

File: src/Components/CardboardListCallout/CardboardListCallout.tsx [hunk 2]
```diff
@@ -124,12 +128,14 @@ const CardboardListCallout = <T extends unknown>({
...
-                        data-testid={dataButtonTestId}
+                        {...(dataButtonTestId && {
+                            'data-testid': dataButtonTestId
+                        })}
                         styles={cardboardListCalloutPrimaryButtonStyles}
                         onClick={() =>
                             primaryActionProps.onPrimaryActionClick(searchText)
                         }
-                        disabled={primaryActionProps.disabled}
+                        disabled={primaryActionProps.disabled ? true : false}
```
Comment (msnyder-msft) [inline src/Components/CardboardListCallout/CardboardListCallout.tsx:138]: nit: you can also just pass `!!primaryActionProps.disabled` if you want to convert it to a boolean.

**Review by msnyder-msft — APPROVED** [review]

File: src/Components/CardboardListCallout/CardboardListCallout.test.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    const { getByTestId } = render(
+        <TestComponent onPrimaryActionClick={testCallback} />
+    );
+
+    const modalComponent = screen.getByTestId('focus-trap'); // take the snapshot of the callout modal component with search text
+    expect(modalComponent).toMatchSnapshot();
+
+    const searchTerm = 'test';
+    const searchBox = getByTestId('search-box');
+    fireEvent.change(searchBox, { target: { value: searchTerm } }); // emulate the user typing input
```
Comment (msnyder-msft) [inline src/Components/CardboardListCallout/CardboardListCallout.test.tsx:102]: did this work better than using the `.type` function on the `userEvent` class? 🤔

Comment (iremgokce) [reply src/Components/CardboardListCallout/CardboardListCallout.test.tsx:102]: yeah unfortunately user.type didn't work: https://github.com/testing-library/user-event/issues/676

File: jest.setup.ts (hunk 1 above)
Comment (ccrowley96) [inline jest.setup.ts:6]: ah love this!

**Review by ccrowley96 — APPROVED** [review]

**Pull request merged**